src/hw_ostc_parser.c

```diff
@@ -1032,12 +1070,20 @@ hw_ostc_parser_samples_foreach (dc_parser_t *abstract, dc_sample_callback_t call
 				parser->gasmix[idx].helium = he;
 				parser->gasmix[idx].type = 0;
 				parser->gasmix[idx].enabled = 1;
-				parser->gasmix[idx].diluent = ccr;
+				if (version == 0x24) {
```

Comment:
@torvalds: This is another case where I was able to confirm for OSTC4 (`0x24`) that the current implementation is broken - but not for the rest of the dive computers using this code path.